src/engine.js

```diff
@@ -0,0 +1,211 @@
+// @flow
+/* eslint-env worker */
+import type { SubView } from "autolayout";
+import type ConstraintBuilder from "./constraint-builder";
+import { View } from "autolayout";
+
+// TODO: find the right webpack configuration to make
+// the worker code standalone from the bundle.
```

Comment:
GLOBAL: It's a great habit to get into the practice of always having a full github issue link in a code comment right here below any `TODO` to make sure you always fully capture technical debt.